pkg/controller/enterprisesearch/config_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package enterprisesearch
+
+import (
+	"testing"
+
+	entsv1beta1 "github.com/elastic/cloud-on-k8s/pkg/apis/enterprisesearch/v1beta1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/settings"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_reuseOrGenerateSecrets(t *testing.T) {
+	type args struct {
+		c    k8s.Client
+		ents entsv1beta1.EnterpriseSearch
+	}
+	tests := []struct {
+		name      string
+		args      args
+		assertion func(*testing.T, *settings.CanonicalConfig, error)
+		wantErr   bool
+	}{
+		{
+			name: "Do not override existing keys",
+			args: args{
+				c: k8s.WrappedFakeClient(
+					&v1.Secret{
+						ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "entsearch-sample-ents-config"},
+						Data: map[string][]byte{
+							ConfigFilename: []byte(existingConfigWithSecureSettings),
+						},
+					},
+				),
+				ents: entsv1beta1.EnterpriseSearch{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "entsearch-sample"},
+				},
+			},
+			assertion: func(t *testing.T, got *settings.CanonicalConfig, err error) {
+				expectedSettings := settings.MustCanonicalConfig(map[string]interface{}{
+					SecretSessionSetting:  "alreadysetsessionkey",
+					EncryptionKeysSetting: []string{"alreadysetencryptionkey1", "alreadysetencryptionkey2"},
+				})
+				assert.Equal(t, expectedSettings, got)
+			},
+		},
+		{
+			name: "Do not override existing encryption keys, create missing session key",
+			args: args{
+				c: k8s.WrappedFakeClient(
+					&v1.Secret{
+						ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "entsearch-sample-ents-config"},
+						Data: map[string][]byte{
+							ConfigFilename: []byte(existingConfig),
+						},
+					},
+				),
+				ents: entsv1beta1.EnterpriseSearch{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "entsearch-sample"},
+				},
+			},
+			assertion: func(t *testing.T, got *settings.CanonicalConfig, err error) {
+				// Unpack the configuration to check that some default secure settings have been generated
```

Comment:
let's not call those "secure settings" which may end up being a different concept. `reusableSettings` better matches the struct?